packages/react/src/components/form/Field/fieldA11Y.ts

```diff
@@ -0,0 +1,85 @@
+export const fieldA11Y = (fieldElement: HTMLElement | null) => {
+  if (!fieldElement) return;
+
+  const elements = new Map<Element, string | null>();
+  const uuid = `:${Math.round(Date.now() + Math.random() * 100).toString(36)}`;
+  let input: Element | null = null;
+
+  const process = (mutations: Partial<MutationRecord>[]) => {
+    const changed: Node[] = [];
+    const removed: Node[] = [];
+
+    // Merge MutationRecords
+    for (const mutation of mutations) {
+      if (mutation.attributeName) changed.push(mutation.target ?? fieldElement);
+      changed.push(...(mutation.addedNodes || []));
+      removed.push(...(mutation.removedNodes || []));
+    }
+
+    // Register elements
+    for (const el of changed)
+      if (isElement(el)) {
+        if (isLabel(el)) elements.set(el, el.htmlFor);
+        else if ('validity' in el) input = el;
```

Comment:
Do we need to check for more here? I think `validity` is also present on `<button>`? Could be a potential edge-case if someone uses `input` and `button` inside `Field`. We know of one use-case of using `HelpText` and `Input`.

Reply:
Good catch! Button can have validity - this is correct - as it can contain "type+name+value" from good old days. What do you think about adding a `&& !(el instance of HTMLButtonElement)` will be sufficient, and move this logic to a `isFormAssociated` utility:
```
cons isFormAssociated = (node: Node): el is Element =>
  isElement(node) && 'validity' in el && !(el instance of HTMLButtonElement);
```

Reply:
Sounds good! A bit unsure of the name, but I can't think of a better one atm 😅

Reply:
Used naming from the standard to attach custom elements to FormApi :) https://webkit.org/blog/13711/elementinternals-and-form-associated-custom-elements/

Reply:
hm a button is associated with a form as well, so we can't do `isFormElement` either.
What about `isFormDataElement`? The elements we care about are elements that would be adding their data to `formData`

Reply:
But maybe just `isInput`, `isInputIsh`, `isInputLike`, `isFormElement` dunno? flisespikk ☺️

Reply:
I also thought of `isFormElement` but then it can get confused by `<form>`. Lets leave it as is :) #flisespikking